File: fsharp/bob/Bob.fs
module Bob

type Bob(msg) =
    let isSilent (s:string)   = s.Trim() = ""
    let hasLetters s          = String.exists System.Char.IsLetter s
    let isShouting s          = hasLetters s && s.ToUpper() = s
    let isQuestion (s:string) = s.EndsWith("?") 
    let otherwise _           = true
    let patterns = [|
        (isSilent,   "Fine. Be that way!");
        (isShouting, "Woah, chill out!");
        (isQuestion, "Sure.");
        (otherwise,  "Whatever.")
        |]

    member this.hey() =
        Array.find (fun (p,_) -> p msg) patterns |> snd

File: csharp/bob/Bob.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

public class Bob
{
    private bool isSilence(String statement)
    {
        return (statement.Trim() == "");
    }

    private bool isShouting(String statement)
    {
        return ((statement.Any(x => char.IsLetter(x))) && (statement.ToUpper() == statement));
    }

    private bool isQuestion(String statement)
    {
        return (statement.EndsWith("?"));
    }

    public String Hey(String statement)
    {
        if (isSilence(statement)) return "Fine. Be that way!";
        if (isShouting(statement)) return "Whoa, chill out!";
        if (isQuestion(statement)) return "Sure.";
        return "Whatever.";
    }
}
